components/selection/test/selection.vdiff.js

```diff
@@ -0,0 +1,313 @@
+					${createInputList(1)}
+				</d2l-test-selection>
+			` },
+			{ name: 'with-pageable', template: html`
+				<d2l-test-pageable>
+					<d2l-selection-controls></d2l-selection-controls>
+					${createInputList(1)}
+				</d2l-test-pageable>
+			` },
+			{ name: 'with-pageable-more', template: html`
+				<d2l-test-pageable item-count="10">
+					<d2l-selection-controls></d2l-selection-controls>
+					${createInputList(1)}
+				</d2l-test-pageable>
+			` },
+			{ name: 'with-selection-pageable', template: html`
+				<d2l-test-selection-pageable>
+					<d2l-selection-controls></d2l-selection-controls>
+					${createInputList(1)}
+				</d2l-test-selection-pageable>
+			` },
+			{ name: 'with-custom-no-selection-text', template: html`
+				<d2l-test-selection-pageable>
+					<d2l-selection-controls no-selection-text="custom no selection text"></d2l-selection-controls>
+					${createInputList(1)}
+				</d2l-test-selection-pageable>
+			` },
+		].forEach(({ name, template }) => {
+			it(name, async() => {
+				const elem = await fixture(template, { viewport });
+				await expect(elem).to.be.golden();
+			});
+		});
+	});
+
+	describe('checkbox', () => {
+		const defaultTemplate = html`<d2l-test-selection><d2l-selection-input label="item 1" key="key1"></d2l-selection-input></d2l-test-selection>`;
+		const selectedTemplate = html`<d2l-test-selection><d2l-selection-input label="item 1" key="key1" selected></d2l-selection-input></d2l-test-selection>`;
+
+		[
+			{ name: 'default', template: defaultTemplate },
+			{ name: 'focus', template: defaultTemplate, action: focusElem },
+			{ name: 'click', template: defaultTemplate, action: clickElem },
+			{ name: 'selected', template: selectedTemplate },
+			{ name: 'skeleton', template: html`<d2l-test-selection><d2l-selection-input label="item 1" key="key1" skeleton></d2l-selection-input></d2l-test-selection>` },
+			{ name: 'selected-focus', template: selectedTemplate, action: focusElem },
+			{ name: 'selected-click', template: selectedTemplate, action: clickElem }
+		].forEach(({ name, template, action }) => {
+			it(name, async() => {
+				const elem = await fixture(template);
+				const input = elem.querySelector('d2l-selection-input');
+				if (action) await action(input);
+				await expect(input).to.be.golden();
+			});
+		});
+	});
+
+	describe('radio', () => {
+		const defaultTemplate = html`<d2l-test-selection selection-single><d2l-selection-input label="item 1" key="key1"></d2l-selection-input></d2l-test-selection>`;
+		const selectedTemplate = html`<d2l-test-selection selection-single><d2l-selection-input label="item 1" key="key1" selected></d2l-selection-input></d2l-test-selection>`;
+
+		[
+			{ name: 'default', template: defaultTemplate },
+			{ name: 'focus', template: defaultTemplate, action: focusElem },
+			{ name: 'click', template: defaultTemplate, action: clickElem },
+			{ name: 'space', template: defaultTemplate, action: elem => sendKeysElem(elem, 'press', 'Space') },
```

Comment:
Does this `await` the `sendKeysElem` in the action?

Reply:
From what I can tell, _yes_. The above and this seem to be equivalent:
```
action: async(elem) => await sendKeysElem(elem, 'press', 'Space')
```

But if I add `{ }`, then no it does not wait:
```
action: elem => { sendKeysElem(elem, 'press', 'Space'); }
```

Reply:
If we'd rather be explicit we can add the `async` and `await`, although I'll probably need to go back and fix up a few other files, I've been using this single function passthrough strategy for a while now lol

Reply:
Ah ok, that's good to know. I think it's OK to use the shorthand now that it's clear what it's doing.